src/emitter.js

```diff
@@ -0,0 +1,279 @@
+		 * Lower values are called first. Defaults to `10`.
+		 */
+		listenTo: function( emitter, event, callback, ctx, priority ) {
+			var emitters, emitterId, emitterInfo, eventCallbacks;
+
+			// _listeningTo contains a list of emitters that this object is listening to.
+			// This list has the following format:
+			//
+			// _listeningTo: {
+			//     emitterId: {
+			//         emitter: emitter,
+			//         callbacks: {
+			//             event1: [ callback1, callback2, ... ]
+			//             ....
+			//         }
+			//     },
+			//     ...
+			// }
+
+			if ( !( emitters = this._listeningTo ) ) {
+				emitters = this._listeningTo = {};
+			}
+
+			if ( !( emitterId = emitter._emitterId ) ) {
+				emitterId = emitter._emitterId = utils.uid();
+			}
+
+			if ( !( emitterInfo = emitters[ emitterId ] ) ) {
+				emitterInfo = emitters[ emitterId ] = {
+					emitter: emitter,
+					callbacks: {}
+				};
+			}
+
+			if ( !( eventCallbacks = emitterInfo.callbacks[ event ] ) ) {
+				eventCallbacks = emitterInfo.callbacks[ event ] = [];
+			}
+
+			eventCallbacks.push( callback );
+
+			// Finally register the callback to the event.
+			emitter.on( event, callback, ctx, priority );
+		},
+
+		/**
+		 * Stops listening for events. It can be usued at different levels:
+		 *
+		 *  * To stop listening to a specific callback.
+		 *  * To stop listening to a specific event.
+		 *  * To stop listening to all events fired by a specific object.
+		 *  * To stop listening to all events fired by all object.
+		 *
+		 * @param {Emitter} [emitter] The object to stop listening to. If omitted, stops it for all objects.
+		 * @param {String} [event] (Requires `emitter`) The name of the event to stop listening to. If omitted, stops it
+		 * for all events from `emitter`.
+		 * @param {Function} callback (Requires `event`) The function be removed from the call list for the give
+		 * `event`.
+		 */
+		stopListening: function( emitter, event, callback ) {
+			var emitters = this._listeningTo;
+			var emitterId = emitter && emitter._emitterId;
+			var emitterInfo = emitters && emitterId && emitters[ emitterId ];
+			var eventCallbacks = emitterInfo && event && emitterInfo.callbacks[ event ];
+
+			// Stop if nothing has been listened.
+			if ( !emitters || ( emitter && !emitterInfo ) || ( event && !eventCallbacks ) ) {
+				return;
+			}
+
+			if ( callback ) {
+				// All params provided. off() that single callback.
+				emitter.off( event, callback );
+			} else if ( eventCallbacks ) {
+				// Only emitter and event provided. off() all callbacks for that event.
+				while ( ( callback = eventCallbacks.pop() ) ) {
+					emitter.off( event, callback );
+				}
+				delete emitterInfo.callbacks[ event ];
+			} else if ( emitterInfo ) {
+				// Only emitter provided. off() all events for that emitter.
+				for ( event in emitterInfo.callbacks ) {
+					this.stopListening( emitter, event );
+				}
+				delete emitters[ emitterId ];
+			} else {
```

Comment:
And what if we do the following:
1. Call listenTo with emitterA
2. Call stopListening with emitterB that we've never listened to

:trollface: ??

As a result we'll stop listening to the emitterA and every other emitter.